src/sentry/integrations/slack/message_builder/notifications/base.py

```diff
@@ -48,22 +66,49 @@ def build(self) -> SlackBlock:
         elif title:  # ie. "ZeroDivisionError",
             first_block_text += f"*{escape_slack_text(title)}*  \n"
 
+        text = self.notification.get_message_description(self.recipient, ExternalProviders.SLACK)
         if text:  # ie. "division by zero", comments
             first_block_text += text
 
+        return first_block_text
+
+    def _get_actions_block(self) -> SlackBlock | None:
+        """
+        Helper method to get the actions block associated with the current Slack action message.
+        If no actions are found, returns None.
+        """
+        actions_block = []
+        actions = self.notification.get_message_actions(self.recipient, ExternalProviders.SLACK)
+        for action in actions:
+            slack_action_message = SlackMessageAction.to_slack_message_action(action)
+            actions_block.append(slack_action_message.get_button())
+
+        if not actions_block:
+            return None
+
+        return {"type": "actions", "elements": actions_block}
+
+    def build(self) -> SlackBlock:
```

Comment:
broke this function down and added helper methods to make it easier to read and understand the flow. We can now also add tests to enforce individual logic and force contracts